corehq/apps/reports/templates/reports/partials/save_reports_modal.html

```diff
@@ -48,6 +48,7 @@ <h3 data-bind="text: modalTitle"></h3>
                                 <option value="last7">{% trans "Last 7 days" %}</option>
                                 <option value="last30">{% trans "Last 30 days" %}</option>
                                 <option value="lastmonth">{% trans "Last month" %}</option>
+                                <option value="lastyear">{% trans "Last year" %}</option>
```

Comment:
don't know off the top of my head if submitting a request like that for a lot of data can impact performance.  Seems like something that could be stress tested.  Let me know if you think this deserves followup.

It would be easy to selectively disable this option - could just pass a variable into the context to omit this option.